### PR TITLE
Add type hints in several files

### PR DESCRIPTION
**Pull request by triple-Mu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Add type hints in several files

## Modification

- Add type hints in mmcv/device/mlu/_functions.py
- Add type hints in mmcv/image/colorspace.py
- Add type hints in mmcv/onnx/info.py
- Add type hints in mmcv/video/processing.py

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmcv/blob/master/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with some of those projects, like MMDet or MMCls.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcv/image/colorspace.py [hunk 1]
```diff
@@ -272,11 +275,11 @@ def ycbcr2bgr(img):
     return out_img
 
 
-def convert_color_factory(src, dst):
+def convert_color_factory(src: str, dst: str) -> Callable:
 
     code = getattr(cv2, f'COLOR_{src.upper()}2{dst.upper()}')
 
-    def convert_color(img):
+    def convert_color(img: Union[np.ndarray, str]) -> np.ndarray:
```
Comment (zhouzaida) [inline mmcv/image/colorspace.py:282]: Can `img` be str here?

File: mmcv/video/processing.py [hunk 2]
```diff
@@ -52,13 +53,13 @@ def convert_video(in_file,
...
-                 out_file,
-                 size=None,
-                 ratio=None,
-                 keep_ar=False,
-                 log_level='info',
-                 print_cmd=False):
+def resize_video(in_file: str,
+                 out_file: str,
+                 size: Optional[tuple] = None,
+                 ratio: Union[tuple, float] = None,
```
Comment (zhouzaida) [inline mmcv/video/processing.py:59]:
```suggestion
                 ratio: Optional[Union[tuple, float]] = None,
```

File: mmcv/image/colorspace.py (hunk 1 above)
Comment (triple-Mu) [reply mmcv/image/colorspace.py]: Fix

File: mmcv/video/processing.py [hunk 3]
```diff
@@ -110,7 +111,9 @@ def cut_video(in_file,
         log_level (str): Logging level of ffmpeg.
         print_cmd (bool): Whether to print the final ffmpeg command.
     """
-    options = {'log_level': log_level}
+    options: Dict[str, Union[str, float]] = {
+        'log_level': log_level
+    }  # type: ignore
```
Comment (zhouzaida) [inline mmcv/video/processing.py:116]:
```suggestion
    options = {'log_level': log_level}
```

**Review by zhouzaida — APPROVED** [review]

File: mmcv/video/processing.py [hunk 4]
```diff
@@ -52,13 +53,13 @@ def convert_video(in_file,
...
-                 out_file,
-                 size=None,
-                 ratio=None,
-                 keep_ar=False,
-                 log_level='info',
-                 print_cmd=False):
+def resize_video(in_file: str,
+                 out_file: str,
+                 size: Optional[tuple] = None,
+                 ratio: Optional[Union[tuple, float]] = None,
```
Comment (teamwong111) [inline mmcv/video/processing.py:59]:
```suggestion
                 ratio: Union[tuple, float, None] = None,
```

**Review by zhouzaida — APPROVED** [review]

**Review by HAOCHENYE — APPROVED** [review]

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**